added translate, scale, their test files, and docs (#99)
Schulich-Ignite/spark: docs/utilities.md

Results in:

![dist demo](img/dist_2.png)

> ### Scale (one parameter)

Reviewer: Inconsistent with the rest of our docs. Can you merge your `Scale (one parameter)` and `Scale (two parameters)` docs into one single section? Take a look at the docs for [colors](https://spark.schulichignite.com/color/#changing-the-color-of-shapes) on our site for examples on how to do this
```suggestion
### Scale
```

Reviewer: Also move these entire sections (scale, translate) to be with the `### Rotate` docs. No point having them separated by `### dist`. 

Personal opinion, make the order in our docs:
1. Translate
2. Rotate
3. Scale

Reviewer: Also preferred if you can quickly mention inside the `rotate` and `scale` descriptions that you can rotate or scale on a point other than the origin by first calling `translate` to change the origin to the new point to rotate from

> Scales the canvas units by x horizontally and vertically. 

Reviewer: Avoid literally re-stating the code in the docs (just in English). Specifically, avoid mentioning the specific parameter names (unless necessary for understanding), that's what you have the `parameters` section for in the first place. In this case, you can just say "Scales the canvas", but it would be preferred to note that this transforms the entire canvas, not just one shape (see the notes on `### Rotate`)

Usage:

```python
scale(x)
```
**Parameters**

> - x: (float) The factor by which the canvas is scaled.

Reviewer: I recommend using a different variable (like `n`) to avoid confusion between `x` units being an arbitrary number vs `x` units being the horizontal transformation. Please also make appropriate changes through the rest of the docs
```suggestion
- n: (float) The factor by which the canvas is scaled.
```

**Example(s):**

> *Scale horizontal and vertical canvas units by a factor of 2*

Reviewer: Nitpick: to me it's easier to understand if you say "doubles" instead of "scales by factor of two". Perks of giving an unambiguous example in case if people are confused by "The factor by which the canvas is scaled" and seeking an example
```suggestion
*Scale the width and height of the canvas units by double*
```

```python hl_lines="6"

> %%ignite

Reviewer: You're highlighting an empty line with this and missing a line break after `%%ignite`` (for consistency). Try 9?
```suggestion
```python hl_lines="9"
%%ignite

```

def setup():
    size(400, 400)

>     circle(100, 100, 100)

Reviewer: This extra circle just complicates the example. Also this one is completely hidden by the scaled up example... (remember to edit the highlight line `hl_lines` accordingly)
```suggestion
```

    fill_style("red")
    
    # apply scale of 2, this will scale canvas units by factor of 2 horizontally and vertically
    scale(2)
    circle(100, 100, 100)
```

Results in:

![scale demo](img/scale_1.png)

> ### Scale (two parameters)

Reviewer: See comments above, delete this section and move the example from here onto the end of the `### Scale` section

Scales the canvas units by x horizontally and by y vertically. 

Usage:

```python

> scale(x,y)

Reviewer: PEP8: Add spaces after your commas
```suggestion
scale(x, y)
```
If in doubt about style, then take a look at the [styling guide](https://docs.google.com/presentation/d/1o1pLhkoDgmy1QIT-70sZVw5N-t9IWiQOCMnyPp-gfhg/edit?usp=sharing) we use for our slide decks

```
**Parameters**

- x: (float) The factor by which the canvas is scaled horizontally.
- y: (float) The factor by which the canvas is scaled vertically.

**Example(s):**

*Scale horizontal and vertical canvas units by factors of 0.75 and 1.25 respectively*

```python hl_lines="6"

> %%ignite

Reviewer: Again, change to 9 and add line break
```suggestion
```python hl_lines="9"
%%ignite

```

def setup():
    size(400, 400)

>     circle(100, 100, 100)

Reviewer: This extra circle just complicates the example (remember to edit the highlight line `hl_lines` accordingly)
```suggestion
```

    fill_style("red")
    
    # apply scale of (0.75, 1.25), this will scale canvas units by factor of 0.75 horizontally and 1.25 vertically
    scale(0.75, 1.25)
    circle(100, 100, 100)
```

Results in:

![scale demo](img/scale_2.png)

### Translate

> Moves the canvas and its origin on the grid. x indicates the horizontal distance to move, and y indicates vertical distance to move.

Reviewer: Don't mention the parameter names (see comments above)
Also I'm confused by "the canvas and its origin on the grid"
* Why separate "canvas and its origin", does this imply you can move the two separately through some other function?
* What's the definition of "the grid"? You can just omit this completely since you never need to reference this "grid" again (otherwise there's burden to give a proper definition)
```suggestion
Change the origin of the canvas.
```

Usage:

```python

> translate(x,y)

Reviewer: PEP8 convention: Add spaces after your commas
```suggestion
translate(x, y)
```

```
**Parameters**

- x: (float) The horizontal distance to translate the canvas.
- y: (float) The vertical distance to translate the canvas.

**Example(s):**

*Translate the canvas 50 units right and 75 units down*

```python hl_lines="6"

> %%ignite

Reviewer: ```suggestion
```python hl_lines="9"
%%ignite

```

def setup():
    size(400, 400)

>     circle(100, 100, 100)

Reviewer: This extra circle just complicates the example (remember to edit the highlight line `hl_lines` accordingly)
```suggestion
```
